.github/workflows/ci.yml

```diff
@@ -116,95 +96,26 @@ jobs:
-        uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: ${{ env.RUST_STABLE_VER }}
-
-      - name: restore cache
-        uses: Swatinem/rust-cache@v2
-
-      - name: cargo test glazier --doc
-        run: cargo test --doc --no-default-features --features=accesskit
-
-  # This tests the future rust compiler to catch errors ahead of time without
-  # breaking CI
-  # We only run on a single OS to save time; this might let some errors go
-  # undetected until the compiler updates and they break CI; but that should
-  # happen rarely, and not pose too much of a problem when it does.
-  test-beta:
-    runs-on: macOS-latest
-    name: cargo clippy+test beta
-    steps:
-      - uses: actions/checkout@v3
-
-      - name: install beta toolchain
-        uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: "beta"
-          components: clippy
-
-      - name: restore cache
-        uses: Swatinem/rust-cache@v2
-
-      - name: cargo clippy glazier
-        run: cargo clippy --all-targets -- -D warnings
-        continue-on-error: true
-
-      - name: cargo test glazier
-        run: cargo test --no-default-features --features=x11
-        continue-on-error: true
-
-  doctest-beta:
-    runs-on: macOS-latest
-    name: glazier doctests beta
-    steps:
-      - uses: actions/checkout@v3
-
-      - name: install beta toolchain
-        uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: "beta"
-
-      - name: restore cache
-        uses: Swatinem/rust-cache@v2
-
-      - name: cargo test glazier --doc
-        run: cargo test --doc --features=accesskit
-
-  check-docs:
-    name: Docs
+  docs:
+    name: cargo doc
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macOS-latest, windows-2019]
+        os: [windows-latest, macos-latest, ubuntu-latest]
     steps:
       - uses: actions/checkout@v3
 
-      - name: install dependencies
+      - name: install additional linux dependencies
         run: |
           sudo apt update
-          sudo apt install libxkbcommon-dev libxkbcommon-x11-dev
+          sudo apt install libwayland-dev libxkbcommon-x11-dev
         if: contains(matrix.os, 'ubuntu')
 
-      - name: install stable toolchain
-        uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: ${{ env.RUST_STABLE_VER }}
+      - name: install nightly toolchain
+        uses: dtolnay/rust-toolchain@nightly
 
       - name: restore cache
         uses: Swatinem/rust-cache@v2
 
-      - name: cargo doc glazier
-        run: cargo doc --no-deps --document-private-items
-
-      # On Linux also attempt docs for X11.
-      - name: cargo doc glazier (X11)
-        run: cargo doc --features=x11 --no-deps --document-private-items
-        if: contains(matrix.os, 'ubuntu')
+      - name: cargo doc
+        run: cargo doc --workspace --all-features --no-deps --document-private-items -Zunstable-options -Zrustdoc-scrape-examples
```

Comment:
This doesn't seem to give any feedback for warnings. Do we want a solution for that?

Reply:
For some warnings, probably. However there are also warnings due to `rustdoc` limitations that we can't do much about. Like when two different structs have the same name and generate a html file name collision.

Although I guess we can maybe enable treating warnings as errors, and then disable them per-repo where we encounter warnings that we won't solve.